LibraryManagement.cpp

```diff
@@ -2,342 +2,332 @@
-            			break;
-        		}
-        		break;
-        			case 2:
-            			cout<<"BOOK\n";
-            			cout<<"[1]ADD BOOK\n";
-            			cout<<"[2]DELETE BOOK\n";
-            			cout<<"[3]EDIT BOOK\n";
-            			cout<<"[4]SEARCH BOOK\n";
-            			cout<<"[5]VIEW ALL BOOKS\n";
-            			cout<<"[6]TOP 3 BOOKS OF THE WEEK\n";
-            			cout<<"[7]EXIT\n";
-            	int choice;
-            		cout<<"Enter selection: ";
-            		cin>>choice;
-            	switch(choice) {
-        			case 1:
-            			bookMark.addBook(bookMark.bookCounter);
-            			break;
-        			case 2:
-            			bookMark.deleteBook(bookMark.bookCounter);
-            			break;
-        			case 3:
-            			bookMark.editBook(bookMark.bookCounter);
-            			break;
-        			case 4:
-            			bookMark.searchBook(bookMark.bookCounter);
-            			break;
-        			case 5:
-            			bookMark.viewAllBooks(bookMark.bookCounter);
-        			case 6:
-            			bookMark.displayTopBooks();
-            			main();
-            			break;
-        			default:
-            			cout<<"Invalid input\n";
-            			break;
-            	}
-
-        			case 3:
-            			quit=true;
-            			break;
-        			default:
-            			cout<<"Invalid input";
-        	}
-        }while(!quit); {
-        		cout<<"Thanks for using our library management";}
-
-
-
-
-	}else if(userType=='R' || userType=='r') {
-		cout<<"Welcome, "<<username<<"!"<<endl;
-		cout<<"You are a regular user."<<endl;
-		cout<<"[1]USER\n";
-		cout<<"[2]BOOK\n";
-		cout<<"[3]EXIT\n";
-		int choice;
-			cout<<"Enter selection: ";
-			cin>>choice;
-		switch(choice) {
+            switch (choose) {
+                case 1:
+                    menu.userOptions();
+                    int selection;
+                    cout << "Enter selection: ";
+                    cin >> selection;
+                    switch (selection) {
+                        case 1:
+                            bookmark->addUser();
+                            break;
+                        case 2:
+                            bookmark->deleteUser();
+                            break;
+                        case 3:
+                            bookmark->editUser();
+                            break;
+                        case 4:
+                            bookmark->viewAllUsers();
+                            break;
+                        case 5:
+                            shouldLeaveApp = true;
+                            break;
+                        default:
+                            cout << "Invalid input\n";
+                            break;
+                    }
+                    break;
+                case 2:
+                    menu.bookOptions();
+                    int choice;
+                    cout << "Enter selection: ";
+                    cin >> choice;
```

Comment:
since you are repeating this multiple times it would be wise to have a separate function for this.
for example:

```
public int askAndGetUserSelection() {
 cout << "Enter selection: ";
 cin >> choice;
}
```